app/src/main/java/com/hwang/xsighting/ViewSighting.java

```diff
@@ -0,0 +1,79 @@
+package com.hwang.xsighting;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.hwang.xsighting.models.Sighting;
+
+public class ViewSighting extends AppCompatActivity {
+
+    //TODO: Check that as 'final' it does not cause issues
+    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private final String sightingId = getIntent().getStringExtra("SIGHTING_ID");
```

Comment:
It makes me nervous that you call getIntent() before onCreate() gets called; I'd rather set the sightingId inside of onCreate because of that.

Reply:
I'd also guess that if you're passing the id of the sighting, you could pass the full details of the sighting here, and not need to make a DB request from this method at all.